polymorphic<T> is not unconditionally copy-constructible

The paper states that while `indirect<T, Alloc>` is copy constructible where `T` is, `polymorphic<T, Alloc>` is unconditionally copy constructible but trying to instantiate `polymorphic<T>` where `T` is not copy constructible fails to compile:
```
struct NonCopyable {
    NonCopyable() = default;
    NonCopyable(const NonCopyable&) = delete;
    NonCopyable& operator=(const NonCopyable&) = delete;
    NonCopyable(NonCopyable&&) = default;
    NonCopyable& operator=(NonCopyable&&) = default;
};

using Pol = xyz::polymorphic<NonCopyable>;    
static_assert(std::copyable<Pol>);

int main() {
    Pol{};
}
```
```
/app/raw.githubusercontent.com/jbcoe/value_types/main/polymorphic.h:64:48: error: use of deleted function 'NonCopyable::NonCopyable(const NonCopyable&)'
   64 |   constexpr direct_control_block(Ts&&... ts) : u_(std::forward<Ts>(ts)...) {
      |                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
<source>:6:5: note: declared here
    6 |     NonCopyable(const NonCopyable&) = delete;
      |     ^~~~~~~~~~~
```
https://godbolt.org/z/44P4YnxsG

--- a/polymorphic.h
+++ b/polymorphic.h
@@ -114,6 +114,7 @@ class polymorphic {
 
   explicit constexpr polymorphic() {
     static_assert(std::is_default_constructible_v<T>);
+    static_assert(std::is_copy_constructible_v<T>);
     using cb_allocator = typename std::allocator_traits<
         A>::template rebind_alloc<detail::direct_control_block<T, T, A>>;
     using cb_traits = std::allocator_traits<cb_allocator>;
@@ -131,6 +132,7 @@ class polymorphic {
   explicit constexpr polymorphic(std::allocator_arg_t, const A& alloc)
       : alloc_(alloc) {
     static_assert(std::is_default_constructible_v<T>);
+    static_assert(std::is_copy_constructible_v<T>);
     using cb_allocator = typename std::allocator_traits<
         A>::template rebind_alloc<detail::direct_control_block<T, T, A>>;
     using cb_traits = std::allocator_traits<cb_allocator>;

